### PR TITLE
[BUGFIX] fixes `--ssl=false` when using `ember test`

### PR DESCRIPTION
**Pull request by linearza**
At the moment when you have the ssl keys defined in `.ember-cli` such as:
```
{
 ...
  "ssl": true,
  "ssl-key": "ssl/server.key.pem",
  "ssl-cert": "ssl/server.crt.pem"
}
```

And you run `ember test --ssl=false` then the key values are still passed through, causing testem to enable SSL - which in turn fails in CI as it looks for the ssl certs

Original change made here: https://github.com/ember-cli/ember-cli/pull/8171
Related issue here: https://github.com/ember-cli/ember-cli/issues/5029#issuecomment-154475311

### CODE REVIEW

File: tests/unit/tasks/test-test.js [hunk 1]
```diff
@@ -107,4 +107,31 @@ describe('test task test', function () {
       sslCert: 'ssl/server.cert',
     });
   });
+
+  it('supports properly disabling SSL by nullifying related keys', function () {
```
Comment (bertdeblock) [inline tests/unit/tasks/test-test.js:111]: This test passes without the code changes as well, which makes it hard to tell if these changes are desired.
Also, the expectations in this test are already covered here:
https://github.com/ember-cli/ember-cli/blob/master/tests/unit/tasks/test-test.js#L55-L56

**Review by bertdeblock — COMMENTED** [review]
I can't seem to reproduce this issue locally.

If I add the following to my `.ember-cli` file:

```
"ssl": true,
"ssl-key": "ssl/server.key.pem",
"ssl-cert": "ssl/server.crt.pem",
```

and run `ember test --ssl=false`, everything works as expected.

**Comment by bertdeblock** [comment]
Closing for now, feel free to reopen if you would like to pick this back up!

**Comment by cyril-sf** [comment]
@bertdeblock I'm running into this issue as well.

In my case, it fails on the CI because the ssh key and cert are not there at all. 

Try to set
```
"ssl": true,
"ssl-key": "ssl/non-existing.key.pem",
"ssl-cert": "ssl/non-existing.crt.pem",
```

and then run `ember test --ssl=false`

**Pull request closed without merging**